docs/design/chart_format.md

```diff
@@ -0,0 +1,664 @@
+﻿# Helm-DM Charts Proposal
+
+
+This document contains a proposal for a new Chart format that both Helm and DM would standardize upon as Helm becomes a DM client.
+
+
+* [Changes](#changes)
+* [tl;dr: Summary](#tldr-summary)
+* [Goals](#goals)
+  * [Non-Goals](#non-goals)
+* [The Chart Format](#the-chart-format)
+  * [Directory Layout](#directory-layout)
+  * [Generators and Templates](#generators-and-templates)
+    * [The Generator Header](#the-generator-header)
+    * [Python/Jinja Templates (aka DM Templates)](#pythonjinja-templates-aka-dm-templates)
+    * [Go Templates (formerly Helm Templates)](#go-templates-formerly-helm-templates)
+  * [The Chart File](#the-chart-file)
+    * [Dependency Information](#dependency-information)
+  * [Releasing a Chart](#releasing-a-chart)
+* [The Chart Repository](#the-chart-repository)
+  * [The Provenance File](#the-provenance-file)
+  * [Repository Protocol](#repository-protocol)
+  * [Aside: Why a Flat Repository Namespace?](#aside-why-a-flat-repository-namespace)
+  * [Aside: Why Not Git(Hub) or VCS?](#aside-why-not-github-or-vcs)
+* [Chart References: Long form, short form, and local reference](#chart-references-long-form-short-form-and-local-reference)
+  * [Long Form](#long-form)
+  * [Short Form](#short-form)
+  * [Local References](#local-references)
+* [Appendix A: User Stories for Charts](#appendix-a-user-stories-for-charts)
+* [Appendix B: The Standard Chart Repository](#appendix-b-the-standard-chart-repository)
+* [Appendix C: EBNF for Generate Header](#appendix-c-ebnf-for-generate-header)
+* [Appendix D: References](#appendix-d-references)
+
+
+## Changes
+
+| Date        | Author        | Changes                                                |
+| ------------|:-------------:|:-------------------------------------------------------|
+| 2016-01-21  | mbutcher@deis | Added manifests/ to chart.                             |
+| 2016-01-25  | mbutcher@deis | Added clarifications based on comments.                |
+| 2016-01-26  | mbutcher@deis | Added hook/, removed manifests/. Added generate header.|
+
+
+## tl;dr: Summary
+* A **Chart** is a folder or archive containing a _Chart.yaml_ file, a _templates/_ directory with one or more template files and supporting files, such as schemas and UI directives, and optionally a _README.md_ and _LICENSE_ file.
+* When a chart is **released**, it is tested, versioned, provenanced, and loaded into a chart repository.
+* A **Chart Repository** is organized according to the conventions of object storage. It is accessible by combining a domain with a repository ID (bucket), and can be browsed. A chart repository contains one or more versioned charts.
```

Comment:
I believe we still hadn't fleshed out how supporting github as repo source would work. There are some comments on the google doc related to it but no resolution. Let me know how you would like to proceed. @jackgr